src/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreMetrics.cs

```diff
@@ -0,0 +1,53 @@
+// <copyright file="AspNetCoreMetrics.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics.Metrics;
+using System.Reflection;
+using OpenTelemetry.Instrumentation.AspNetCore.Implementation;
+
+namespace OpenTelemetry.Instrumentation.AspNetCore
+{
+    /// <summary>
+    /// Asp.Net Core Requests instrumentation.
+    /// </summary>
+    internal class AspNetCoreMetrics : IDisposable
+    {
+        internal static readonly AssemblyName AssemblyName = typeof(HttpInListener).Assembly.GetName();
```

Comment:
Need to be defensive here in case any exception happened https://social.msdn.microsoft.com/Forums/windowsapps/en-US/ba19700e-8f63-411d-9097-63a37c7e8a4d/assemblygetname-throws-securityexception?forum=netfxtoolsdev.

Reply:
Not blocking this PR, in general we might want to be defensive while having static initialization logic - `TypeInitializationException` could be very hard to troubleshoot.

Reply:
Yea good point. This should be fixed elsewhere as well - here's another example https://github.com/open-telemetry/opentelemetry-dotnet/blob/a4d8b3509437497413b9b70f4cda4dda55b7bff9/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L31-L36.

I can grab this in a follow up.